app/src/main/java/com/example/dine_and_donate/SignUpActivity.java

```diff
@@ -48,52 +48,54 @@ protected void onCreate(Bundle savedInstanceState) {
+        mSignUpBtn = findViewById(R.id.final_signup_btn);
+        mBackToLogin = findViewById(R.id.back_to_login_btn);
+
+        mName.setVisibility(View.GONE);
+        mEmail.setVisibility(View.GONE);
+        mPassword.setVisibility(View.GONE);
+        mOrgPhone.setVisibility(View.GONE);
+        mSignUpBtn.setVisibility(View.GONE);
 
-        name.setVisibility(View.GONE);
-        email.setVisibility(View.GONE);
-        password.setVisibility(View.GONE);
-        orgPhone.setVisibility(View.GONE);
-        signUpBtn.setVisibility(View.GONE);
 
         //display specific text views depending on user type selected
-        spinner = findViewById(R.id.user_options);
+        mSpinner = findViewById(R.id.user_options);
         ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.user_types, android.R.layout.simple_spinner_item);
         adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-        spinner.setAdapter(adapter);
+        mSpinner.setAdapter(adapter);
 
         //show appropriate text views based on user type selection
-        spinner.setOnItemSelectedListener(new Spinner.OnItemSelectedListener() {
+        mSpinner.setOnItemSelectedListener(new Spinner.OnItemSelectedListener() {
             public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                 String selectedItem = parent.getSelectedItem().toString();
                 //show all fields except phone number for consumer user type
                 if (selectedItem.equals("Consumer")) {
-                    name.setVisibility(View.VISIBLE);
-                    email.setVisibility(View.VISIBLE);
-                    password.setVisibility(View.VISIBLE);
-                    signUpBtn.setVisibility(View.VISIBLE);
-                    orgPhone.setVisibility(View.GONE);
-                }  else if(spinner.getSelectedItem().toString().equals("Organization")) {
-                    orgPhone.setVisibility(View.VISIBLE);
-                    name.setVisibility(View.VISIBLE);
-                    email.setVisibility(View.VISIBLE);
-                    signUpBtn.setVisibility(View.VISIBLE);
-                    password.setVisibility(View.VISIBLE);
+                    mName.setVisibility(View.VISIBLE);
+                    mEmail.setVisibility(View.VISIBLE);
+                    mPassword.setVisibility(View.VISIBLE);
+                    mSignUpBtn.setVisibility(View.VISIBLE);
+                    mOrgPhone.setVisibility(View.GONE);
+                }  else if(mSpinner.getSelectedItem().toString().equals("Organization")) {
+                    mOrgPhone.setVisibility(View.VISIBLE);
+                    mName.setVisibility(View.VISIBLE);
+                    mEmail.setVisibility(View.VISIBLE);
+                    mSignUpBtn.setVisibility(View.VISIBLE);
+                    mPassword.setVisibility(View.VISIBLE);
                 } else {
```

Comment:
is this case possible?

Reply:
if not, you only need to change the visibility of mOrgPhone, which would simplify this code a bunch

Reply:
if the user goes from the "Consumer" to empty spinner selection, then none of the edit text views should be on the page.